### PR TITLE
Add old upgrade guide, update it and integrate with new chapters

### PR DESCRIPTION
**Pull request by pbauer**

### CODE REVIEW

**Comment by mauritsvanrees** [comment]
Thanks @pbauer !
I have started looking at this, and pushed two small changes.
Locally with `make livehtml` I see a few warnings about missing links, images and references. Do you have some uncommitted files locally? Especially `backend/upgrading/addon_upgrade`.

```
reading sources... [100%] backend/upgrading/version_specific_migration/upgrade_to_51                                    
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/index.md:21: WARNING: toctree contains reference to nonexisting document 'backend/upgrading/addon_upgrade'
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:312: WARNING: image file not readable: _static/plip_hidpi_low_resolution.png
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:319: WARNING: image file not readable: _static/plip_hidpi_high_resolution.png
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:378: WARNING: image file not readable: _static/site-control-panel-icon-thumb-settings.png
looking for now-outdated files... none found
pickling environment... done
checking consistency... /Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/volto/cheatsheet.md: WARNING: document isn't included in any toctree
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/volto/releases.md: WARNING: document isn't included in any toctree
done
preparing documents... done
writing output... [100%] index                                                                                          
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:17: WARNING: unknown document: /develop/addons/upgrade_to_51
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:61: WARNING: undefined label: resource_bundle_aggregation
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:77: WARNING: undefined label: create_actions_ttw
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:134: WARNING: undefined label: addon_installation_code
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:256: WARNING: undefined label: using_external_catalogs
/Users/maurits/community/plone-coredev/6.0/documentation/docs/docs/backend/upgrading/version_specific_migration/upgrade_to_51.md:385: WARNING: undefined label: content-type-icons-changed
```

**Comment by mauritsvanrees** [comment]
I have pushed some more fixes to Preparation and Troubleshooting.

I wonder if we should include the old version specific information. We could include only the upgrade info from 5.2 to 6.0, and point to the old docs for older versions. But we should be sure that the old information remains available under some url. If the best way to ensure this is by including it here, that is fine with me.

The old docs have a page in `develop/addons` with specific information on [upgrading an add-on to 5.1](https://docs.plone.org/develop/addons/upgrade_to_51.html). I don't know why this was only done for 5.1, and why this is not in the main upgrade guide. We should probably merge this here.

**Comment by stevepiercy** [comment]
I'm also working on this guide. I'll pull in the commits from @mauritsvanrees, resolve, incorporate his comments, and push my work tonight. `make livehtml` reveals many b0rken things.

File: docs/backend/upgrading/preparations.md [hunk 1]
```diff
@@ -16,54 +16,45 @@ Things to do before you migrate Plone.
...
-- Read the following files in the CMFPlone directory of the distribution of the new version of Plone you want to update to:
-
-  - README.txt
-
-  - INSTALL.txt
-
-  - UPGRADE.txt (although this usually contains only the general procedure outlined above)
-
-    - These files are important because they may contain important last minute information and might be more specific than the relevant sections of this reference manual.
-
```
Comment (stevepiercy) [inline docs/backend/upgrading/preparations.md]: I could not find these files anywhere in https://dist.plone.org/release/6.0.0b2/, so I agree this is obsolete.

File: docs/backend/upgrading/preparations.md [hunk 2]
```diff
@@ -16,54 +16,45 @@ Things to do before you migrate Plone.
...
   - Make sure all the add-on products you are using have updated to support the version of Plone you are upgrading to.
 
   - Start with the third-party products that are in use on your site.
+    Verify that they have been updated or verified to work on the new version, and get them upgraded in your existing instance before you start the Plone/Zope/Python upgrade if possible.
 
-    - Verify that they have been updated or verified to work on the new version, and get them upgraded in your existing instance before you start the Plone/Zope/Python upgrade if possible.
-
-  - If Zope depends on a newer version of Python, install the new version of Python first.
-
-  - If the newer version of Plone depends on a newer version of Zope, you will need to install that before proceeding with the Plone upgrade.
```
Comment (stevepiercy) [inline docs/backend/upgrading/preparations.md]: I agree that we should not duplicate documentation of Zope, and together with the updated note providing better references to migration, I like this change.

**Comment by pbauer** [comment]
I think having a upgrade-guide in the current docs that has info on the old versions and not only 5.2 to 6.0 is a good thing. Thus I added the old addon-upgrade-docs as well. But I don't want to force it so if you disagree feel free to remove these.

**Comment by pbauer** [comment]
I think this can be merged for now.

**Comment by stevepiercy** [comment]
Please wait until I complete my review. There is no rush. I have been kinda busy working on the installation docs, which take precedent.

File: docs/backend/upgrading/add-on-upgrade.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+```
+
+- Run *bin/buildout*. Wait until all new software is downloaded and installed.
+- Start Plone again - your site may look weird, or even be inaccessible until you have performed the next step
+- Navigate to the Add-on screen (add `/prefs_install_products_form` to your site URL, and upgrade products if you can (products that support both your current and new version of Plone).
+  : - Perform product-specific upgrade procedures (if any).
+    - You will find these in the documentation of each product.
+
+Should the `/prefs_install_products_form` be unreachable, you should try doing the add-on upgrades from the Management Interface.
+Navigate to the quickinstaller in the Management Interface, and reinstall or upgrade products that are shown to be outdated.
```
Comment (stevepiercy) [inline docs/backend/upgrading/add-on-upgrade.md]: Where is this, "quickinstaller in the Management Interface"? I cannot find it anywhere in the ZMI.

**Review by stevepiercy — COMMENTED** [review]
@pbauer @mauritsvanrees or anyone else know where is "quickinstaller in the Management Interface"?

**Comment by stevepiercy** [comment]
While reviewing content in this PR, I am discovering that there is incorrect and outdated information. I am adamant about publishing only correct and current information for Plone 6 documentation. That means reviews will take longer than expected, yet the content will be of higher quality. The steps of migrating docs by copy-paste and getting the docs to build without warnings and errors are always necessary and important, yet do not address this matter of quality.

File: docs/backend/upgrading/version-specific-migration/upgrade-to-python3.md [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+
+In the GitHub repository of the add-on:
+
+- Open a ticket with the title "Add support for Python 3".
+- Create a new branch named `python3`.
+
+```{warning}
+The following section is valid until the final release of Plone 5.2.
+Upon the final release of Plone 5.2, something else will take its place.
+```
```
Comment (stevepiercy) [inline docs/backend/upgrading/version-specific-migration/upgrade-to-python3.md]: Can this warning be removed or replaced with whatever that "something" is?

**Review by stevepiercy — COMMENTED** [review]
@pbauer @mauritsvanrees or anyone else know where is "quickinstaller in the Management Interface"?

File: docs/backend/upgrading/version-specific-migration/upgrade-to-python3.md (hunk 4 above)
Comment (mauritsvanrees) [reply docs/backend/upgrading/version-specific-migration/upgrade-to-python3.md]: The warning can be removed. I have no idea what to replace it with, and the documentation below seems okay still.

**Comment by mauritsvanrees** [comment]
> @pbauer @mauritsvanrees or anyone else know where is "quickinstaller in the Management Interface"?

In Plone 5.2 and earlier it is in `portal_quickinstaller`.
In Plone 6.0, this tool is no longer there and its code is not included. One of the earliest upgrade steps in Plone 6.0 removes the tool if it is still there.

File: docs/backend/upgrading/version-specific-migration/upgrade-to-python3.md (hunk 4 above)
Comment (stevepiercy) [reply docs/backend/upgrading/version-specific-migration/upgrade-to-python3.md]: Removed in a7076aa98989067b994219c7884d99f29287adff

**Comment by stevepiercy** [comment]
> In Plone 5.2 and earlier it is in `portal_quickinstaller`. In Plone 6.0, this tool is no longer there and its code is not included. One of the earliest upgrade steps in Plone 6.0 removes the tool if it is still there.

At https://demo.plone.org/portal_quickinstaller/manage_installProductsForm:

> WARNING: This form is deprecated. Please use the [Add-ons control panel](https://demo.plone.org/prefs_install_products_form) in the Plone UI.

The wording of this will be a little tricky, but let me see what I can whip together.

File: docs/backend/upgrading/add-on-upgrade.md (hunk 3 above)
Comment (stevepiercy) [reply docs/backend/upgrading/add-on-upgrade.md]: @mauritsvanrees would you please take a look at commit ab70d94633b2e64e619f91f4b3c89d31a1baa124, and provide any feedback? Thank you!

Comment (mauritsvanrees) [reply docs/backend/upgrading/add-on-upgrade.md]: Yes, that commit looks fine. Thanks.

**Comment by stevepiercy** [comment]
I have completed my review. I fixed mostly English and MyST syntax and grammar, some links and redirects, and a bunch of nit-picky details.

Now is the time for a final review. @pbauer @ksuess @mauritsvanrees

**Review by stevepiercy — DISMISSED** [review]

**Comment by pbauer** [comment]
LGTM

**Review by stevepiercy — APPROVED** [review]
My changes were approved by @pbauer (one cannot approve a PR one creates).

**Pull request merged**